docs/cheat-sheets/javascript-code-injection.mdx

```diff
@@ -0,0 +1,82 @@
+---
+slug: javascript-code-injection
+---
+
+import LinkToRegistryRule from "/src/components/LinkToRegistryRule"
+import CodeInjectionIntro from "/src/components/concept/_code-injection-intro.mdx"
+
+# Code injection prevention for JavaScript
+
+<CodeInjectionIntro />
+
+## 1. Executing or evaluating code
+
+### 1.A. running code with `vm` module
+
+The `vm` module enables compiling and running code within V8 Virtual Machine contexts. The `vm` module is not a security mechanism. Do not use it to run untrusted code. If code passed to vm functions is controlled by user input it could result in code injection. 
+
+Example:
+
+```javascript
+const vm = require('vm');
+const contextObject = { globalVar: 1 };
+
+// safe
+vm.runInContext('globalVar *= 2;', contextObject);
+
+// vulnerable 
+let userInput = 'this.constructor.constructor("return process.env")()'; // Value supplied by user input
+vm.runInContext(`globalVar = ${userInput};`, contextObject);
+```
+
+```javascript
+const vm = require('vm');
+// safe
+const code = `return 'hello ' + name`
+vm.compileFunction(code, [], { parsingContext: vm.createContext({ name: 'name' }) })
+
+// vulnerable
+let userInput = '1; while (true) {}'  // Value supplied by user input
+const code = `let x = ${userInput}; return 'hello ' + x`
+vm.compileFunction(code, [], { parsingContext: vm.createContext({ name: 'name' }) })
+```
```

Comment:
```suggestion
```javascript
const vm = require('vm');
const contextObject = { globalVar: 1 };

// safe
vm.runInContext('globalVar *= 2;', contextObject);

// vulnerable
let userInput = 'this.constructor.constructor("return process.env")()'; // Value supplied by user input
vm.runInContext(`globalVar = ${userInput};`, contextObject);

// safe
const code = `return 'hello ' + name`
vm.compileFunction(code, [], { parsingContext: vm.createContext({ name: 'name' }) })

// vulnerable
let userInput = '1; while (true)
```
```